### PR TITLE
Add entry url validation for unzip utility. Defends against Zip Slip

### PR DESCRIPTION
**Pull request by cdhoffmann**
Changes to validate that the entries within a zip archive have valid paths. Defends against Zip Slip attack.

### CODE REVIEW

File: AEPServices/Sources/utility/unzip/FileUnzipper.swift [hunk 1]
```diff
@@ -58,4 +58,23 @@ public class FileUnzipper: Unzipping {
...
+    ///
+    /// Validates the entryUrl path
+    /// - Parameters:
+    ///     - entryUrl: The url for the entry being extracted from the archive
+    ///     - destination: The destination url for the entry being extracted
+    /// - Returns: True if valid, false if invalid
+    private func isValidPath(_ entryUrl: URL, destination: URL) -> Bool {
+        let destinationComponents = canonicalize(destination).pathComponents
+        let entryComponents = canonicalize(entryUrl).pathComponents
+        return destinationComponents.count < entryComponents.count && !zip(destinationComponents, entryComponents).contains(where: !=)
```
Comment (shalehaha) [inline AEPServices/Sources/utility/unzip/FileUnzipper.swift:70]: interesting, learnt a new method `zip`

**Review by shalehaha — APPROVED** [review]
Looks good to me. @cdhoffmann have you manually tested this change?

**Comment by cdhoffmann** [comment]
> Looks good to me. @cdhoffmann have you manually tested this change?

The test i wrote uses the same zip file found in the android tests. It has an entry in the archive named "../Makefile". If you breakpoint through the test you can see it. Lmk if there are other manual tests you think i should do.

**Review by yangyansong-adbe — APPROVED** [review]

**Comment by jiabingeng** [comment]
@cdhoffmann it'd be better if you can quickly test it on a physical device.

**Comment by cdhoffmann** [comment]
> @cdhoffmann it'd be better if you can quickly test it on a physical device.

@shalehaha Verified on iPhone 12 Pro iOS 14.2.1

**Pull request merged**